tests/kernel/spinlock/testcase.yaml

```diff
@@ -0,0 +1,3 @@
+tests:
+  test:
+    platform_whitelist: esp32
```

Comment:
this doesn't scale.
Think we need something like CONFIG_ARCH_HAS_SMP and filter on that instead.
Or if we assume any board that has SMP will have it enabled by default, filter on CONFIG_SMP.

Reply:
Certainly don't disagree, but this is how this is done with the current test rig.  Tests for particular harware-specific APIs use platform or arch whitelists to select what they want.